### PR TITLE
[mock_uss] avoid full file scan when querying interaction log

### PR DESCRIPTION
**Pull request by Shastick**
This PR attempts to improve the performance of the `/mock_uss/interuss_logging/logs` GET endpoint. It:
 - updates the mock USS's interaction logger to store a full timestamp of the interaction in the relevant filename 
 - updates the endpoint logic to filter out files that are older than the required time
 
 Additionally, returned interactions are now sorted before they are returned.
 
 Test plan: the qualifier is still able to run locally against an updated mock_uss, and the scenarios that rely on the `interuss_logging/logs`  endpoint are passing.
 
 This is not guaranteed to solve #677 but is very likely to improve the situation.

### CODE REVIEW

**Comment by Shastick** [comment]
Some relevant observations I made while investigating scenarios that failed with the new version of the interaction logging endpoint, and the reason for adding an additional time buffer in requests(*)

At least on my local setup, there is a slight clock skew between running containers and my OS: under certain circumstances (which happens to be often or always for me), the mock_uss containers runs slightly behind. As a result interactions are timestamped with a too old timestamp and are ignored when qualifier specifies the time since which interactions should be returned.

Here is a sample of the logged interactions:

```
julien@Woosh scdsc_a_interaction_logs % ls -l
total 64
-rw-r--r--  1 julien  staff  2589 Aug 16 10:19 000000_Outgoing_POST_2024-08-16T08:19:49.217310Z.json
-rw-r--r--  1 julien  staff  2058 Aug 16 10:19 000001_Incoming_GET_2024-08-16T08:19:49.229415Z.json
-rw-r--r--  1 julien  staff  2590 Aug 16 10:19 000002_Outgoing_POST_2024-08-16T08:19:49.355719Z.json
-rw-r--r--  1 julien  staff  2059 Aug 16 10:19 000003_Incoming_GET_2024-08-16T08:19:49.366266Z.json
-rw-r--r--  1 julien  staff  2047 Aug 16 10:19 000004_Incoming_GET_2024-08-16T08:19:49.398353Z.json
-rw-r--r--  1 julien  staff  2962 Aug 16 10:19 000005_Incoming_POST_2024-08-16T08:19:49.418825Z.json
-rw-r--r--  1 julien  staff  1856 Aug 16 10:19 000006_Incoming_POST_2024-08-16T08:19:56.497142Z.json
-rw-r--r--  1 julien  staff  1534 Aug 16 10:19 000007_Outgoing_POST_2024-08-16T08:19:56.517779Z.json
```

The relevant interaction here is one of the two first `Outgoing_POST` ones, which both have a timestamp that is slightly before the time that the qualifier will use, as illustrated in the snapshot below:

Screenshot 2024-08-16 at 10 34 28

(The timestamp is passed in the `from_time` in this query `http://scdsc.log.uss6.localutm/mock_uss/interuss_logging/logs?from_time=2024-08-16T08:19:49.377425Z`)

`08:19:49.377425` is just after the second outgoing request that was made at `08:19:49.355719`

(*) we could consider adding this buffer as a default in the library code that queries the interactions, instead of leaving that responsibility with the caller: opinions welcome. Note that we already decided against having this buffer on the server side, as we want our mock_uss to implement precise APIs.

File: monitoring/uss_qualifier/scenarios/astm/utm/subscription_notifications/test_steps/validate_notification_received.py [hunk 1]
```diff
@@ -150,7 +150,7 @@ def is_applicable(interaction: Interaction) -> bool:
 
 def _check_notification_sent_with_subscription_id_and_response(
     interactions: List[Interaction], subscription_id: str
-) -> Tuple[bool, bool, int]:
+) -> Tuple[bool, int]:
```
Comment (Shastick) [inline monitoring/uss_qualifier/scenarios/astm/utm/subscription_notifications/test_steps/validate_notification_received.py:153]: This is not directly related to the PR, but I found the incorrect type annotation while checking how scenarios validate interactions.

File: monitoring/mock_uss/interaction_logging/routes_interactions_log.py [hunk 2]
```diff
@@ -33,33 +37,39 @@ def interaction_logs() -> Tuple[str, int]:
...
 
+    # individual interactions are logged to a file of the form <index>_<direction>_<method>_<timestamp>.json, eg
+    # 000001_Incoming_GET_2023-08-30T20-48-21.900000Z.json
     interactions: List[Interaction] = []
+
     for fname in os.listdir(log_path):
+        # Parse the interaction time from the file name:
+        fname_parts = fname.split("_")
+        if len(fname_parts) != 4:
+            # TODO: should we consider failing hard here, to let users know something is likely very wrong?
```
Comment (BenjaminPelletier) [inline monitoring/mock_uss/interaction_logging/routes_interactions_log.py:48]: If the folder were necessarily always for the exclusive use of these logs then yes, failing here would probably make sense.  But I don't think we want to document and enforce that -- we can tolerate other content just fine by ignoring it.  The warning helps prompt the user in case they didn't intend that.

**Review by BenjaminPelletier — APPROVED** [review]
Looks good, just two small issues before merging

**Comment by Shastick** [comment]
Both comments addressed.

**Review by BenjaminPelletier — APPROVED** [review]
